### DISCUSSION TITLE
middleware type

### DISCUSSION THREAD

**Discussion by me-v2** [fastapi/fastapi#11349] [Questions]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

app.add_middleware(GZipMiddleware),  pycharm 2023.3.5, will tooltip:: Expected type 'Type[_MiddlewareClass]' (matched generic type 'Type[_MiddlewareClass[ParamSpec("P")]]'), got 'Type[GZipMiddleware]' instead 
![CleanShot 2024-03-26 at 09 22 35](https://github.com/tiangolo/fastapi/assets/6077601/dbe3c863-35ce-45bc-8025-fb48d8bdcf51)

**Comment by YuriiMotov** [comment]
Duplicate with answer: https://github.com/fastapi/fastapi/discussions/10968
